plugins/main/server/health-check/alerting-monitors.ts

```diff
@@ -0,0 +1,276 @@
+    return total > 0;
+  } catch (error) {
+    const _error = error as Error;
+    // If search fails (e.g., missing permissions/index), assume not exists
+    ctx.logger.warn(
+      `monitorExists error for [${name}]: ${_error?.message || _error}`,
+    );
+    return false;
+  }
+}
+
+async function ensureMonitor(
+  ctx: PluginTaskRunContext,
+  sample: SampleMonitorDef,
+  availableDefaultChannelIds?: Set<string>,
+) {
+  const { monitorName, channelId, message, severity } = sample;
+  const exists = await monitorExists(ctx, monitorName);
+  if (exists) {
+    ctx.logger.info(`Monitor already exists [${monitorName}]`);
+    return;
+  }
+
+  // Enforce: do not create sample monitors without their required channel
+  if (
+    !availableDefaultChannelIds ||
+    !availableDefaultChannelIds.has(channelId)
+  ) {
+    ctx.logger.info(
+      `Skipping sample monitor [${monitorName}] because required channel [${channelId}] is not present`,
+    );
+    return;
+  }
+
+  try {
+    await request(ctx, {
+      method: 'POST',
+      path: '/_plugins/_alerting/monitors',
+      querystring: { refresh: 'wait_for' },
+      body: buildMonitorBody(monitorName, severity, channelId, message),
+    });
+    ctx.logger.info(`Created sample monitor [${monitorName}]`);
+  } catch (error) {
+    const _error = error as Error;
+    // Don’t break health check; surface a clear warning
+    ctx.logger.warn(
+      `Could not create sample monitor [${monitorName}]: ${
+        _error?.message || _error
+      }`,
+    );
+  }
+}
+
+export const createSampleAlertingMonitors = async (
+  ctx: PluginTaskRunContext,
+  options?: { availableDefaultChannelIds?: Set<string> },
+) => {
+  try {
+    ctx.logger.info('Starting Alerting sample monitors check');
+    const availableDefaultChannelIds: Set<string> | undefined =
+      options?.availableDefaultChannelIds
+        ? new Set(options.availableDefaultChannelIds)
+        : undefined;
+
+    /**
+     * Sample monitor creation attempts
+     * `await Promise.all(SAMPLES.map(sample => ensureMonitor(ctx, sample)));`
+     *
+     * Throws unhandled promise rejection on logger calls:
+     * [error][data][opensearch] [alerting_exception]: all shards failed
+     * [warning][healthcheck][XXXXXX] Could not create sample monitor [Sample: Jira]: alerting_exception: [alerting_exception] Reason: all shards failed
+     * [error][data][opensearch] [alerting_exception]: all shards failed
+     * [warning][healthcheck][XXXXXX] Could not create sample monitor [Sample: Slack]: alerting_exception: [alerting_exception] Reason: all shards failed
+     */
+    for (const sample of SAMPLES) {
+      await ensureMonitor(ctx, sample, availableDefaultChannelIds);
+    }
+
+    ctx.logger.info('Alerting sample monitors check finished');
+  } catch (error) {
+    const _error = error as Error;
+    const message = `Error ensuring sample monitors: ${
+      _error?.message || _error
+    }`;
+    ctx.logger.warn(message);
```

Comment:
Why a warning instead of an error?

```suggestion
    ctx.logger.error(message);
```